### PR TITLE
Add vector Poisson test with boundary integral

### PR DESCRIPTION
**Pull request by thoeschler**

### CODE REVIEW

File: sandbox/test_vector_poisson.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
+from shenfun import FunctionSpace, Array, TrialFunction, TestFunction, \
+    TensorProductSpace, comm, inner, grad, VectorSpace, BlockMatrix
+import sympy as sp
+import numpy as np
+
+x, y = sp.symbols("x, y")
+
+# manufactured solution with:
+# x - component:
+# ux'(x = - 1) != 0, ux(x = 1) = 0, ux(y = - 1) = 0, ux(y = 1) = 0
```
Comment (sebglane) [inline sandbox/test_vector_poisson.py]: Misleading statement `!=`. I guess `=` is correct.

File: sandbox/test_vector_poisson.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+# y - component:
+# uy(x = - 1) = 0, uy(x = 1) = 0, ux(y = - 1) = 0, ux'(y = 1) != 0
```
Comment (sebglane) [inline sandbox/test_vector_poisson.py]: Same here.

**Review by sebglane — APPROVED** [review]

File: sandbox/test_vector_poisson.py (hunk 2 above)
Comment (sebglane) [reply sandbox/test_vector_poisson.py]: Fixed in b1dbee3.

File: sandbox/test_vector_poisson.py (hunk 1 above)
Comment (sebglane) [reply sandbox/test_vector_poisson.py]: Fixed in b1dbee3.
